src/commands/Minion/pickpocket.ts

```diff
@@ -63,91 +67,128 @@ export default class extends BotCommand {
-		);
-
-		if (!pickpocketable) {
+		)
+			? Pickpocketables.find(
+					npc => stringMatches(npc.name, name) || npc.alias?.some(alias => stringMatches(alias, name))
+			  )
+			: Stalls.find(
+					stall => stringMatches(stall.name, name) || stall.alias?.some(alias => stringMatches(alias, name))
+			  );
+
+		if (!stealable) {
 			return msg.channel.send(
-				`That is not a valid NPC to pickpocket, try pickpocketing one of the following: ${Pickpocketables.map(
+				`That is not a valid NPC/Stall to pickpocket or steal from, try pickpocketing or stealing from one of the following: ${Pickpocketables.map(
 					npc => npc.name
-				).join(', ')}.`
+				).join(', ')}, ${Stalls.map(stall => stall.name).join(', ')}.`
 			);
 		}
 
-		if (pickpocketable.qpRequired && msg.author.settings.get(UserSettings.QP) < pickpocketable.qpRequired) {
+		const isPickpockable: boolean = (<Pickpockable>stealable).slope ? true : false;
+
+		const maxTripLength = msg.author.maxTripLength('Pickpocket');
+
+		if (stealable.qpRequired && msg.author.settings.get(UserSettings.QP) < stealable.qpRequired) {
 			return msg.channel.send(
-				`You need atleast **${pickpocketable.qpRequired}** QP to pickpocket a ${pickpocketable.name}.`
+				`You need atleast **${stealable.qpRequired}** QP to ${isPickpockable ? 'pickpocket' : 'steal from'} a ${
+					stealable.name
+				}.`
 			);
 		}
 
 		if (
-			pickpocketable.itemsRequired &&
-			!bankHasAllItemsFromBank(msg.author.allItemsOwned().bank, pickpocketable.itemsRequired)
+			stealable.itemsRequired &&
+			!bankHasAllItemsFromBank(msg.author.allItemsOwned().bank, stealable.itemsRequired)
 		) {
 			return msg.channel.send(
-				`You need these items to pickpocket this NPC: ${new Bank(pickpocketable.itemsRequired)}.`
+				`You need these items to ${
+					isPickpockable ? 'pickpocket this NPC' : 'steal from this stall'
+				}: ${new Bank(stealable.itemsRequired)}.`
 			);
 		}
 
-		if (msg.author.skillLevel(SkillsEnum.Thieving) < pickpocketable.level) {
+		if (msg.author.skillLevel(SkillsEnum.Thieving) < stealable.level) {
 			return msg.channel.send(
-				`${msg.author.minionName} needs ${pickpocketable.level} Thieving to pickpocket a ${pickpocketable.name}.`
+				`${msg.author.minionName} needs ${stealable.level} Thieving to ${
+					isPickpockable ? 'pickpocket' : 'steal from'
+				} a ${stealable.name}.`
 			);
 		}
 
-		const timeToPickpocket = (pickpocketable.customTickRate ?? 2) * 600;
+		const [hasFavour, requiredPoints] = gotFavour(msg.author, Favours.Hosidius, 15);
+		if (!hasFavour && stealable.name === 'Fruit stall') {
+			return msg.channel.send(
+				`${msg.author.minionName} needs ${requiredPoints}% Hosidius Favour to steal fruit from the Fruit stalls!`
+			);
+		}
 
-		const maxTripLength = msg.author.maxTripLength('Pickpocket');
+		const timeToTheft = isPickpockable
+			? ((<Pickpockable>stealable).customTickRate ?? 2) * 600
+			: (<Stall>stealable).respawnTime;
```

Comment:
we dont do stuff like this anywhere. you should convey this in the type itself. rough example:
```ts
type Stealable = { type: 'pickpocket' | 'stall' } & (PickpocketNPC | Stall);
type Pickpocketable {
type: 'pickpocket',
customTickRate: 1,
}
type Stall {
type: 'stall'
}

const timeToTheft = stealable.type === 'pickpocket' ? stealable.customTickrate : stealable.whatever;
```